### PR TITLE
Tabs integration

### PR DESCRIPTION
**Pull request by mcalus3**
- Fixed minor visual issues
- removed "settings" component, moved travis project slug to annotation and api key to backend proxy.
- simplified state management (now only entity is held in react context and local storage)

Loom link:
https://www.loom.com/share/54fc1ac4aa2647f8999733fee84ae657

### CODE REVIEW

**Comment by dtuite** [comment]
@mcalus3 is this ready for review yet? The build is failing and I see `console.log` statements in the code still. Might be a good idea to mark it as a draft until you feel it is ready for others to have a look.

**Comment by mcalus3** [comment]
> 
> 
> @mcalus3 is this ready for review yet? The build is failing and I see `console.log` statements in the code still. Might be a good idea to mark it as a draft until you feel it is ready for others to have a look.

The code is fixed, but i don't know why `yarn install --frozen-lockfile` on the server doesn't work. Locally on my environment it's working.

EDIT:Changed the github actions so it works now.

**Comment by dtuite** [comment]
@mcalus3  Collecting some of the feedback from the review meeting:

 1. No need to put any effort into supporting multiple CI/CD systems at once.
 2. The column widths need some work.
 3. Support for travis-ci.org should be dropped.
 4. The "All builds" header and support button should stay, at least until we get some guidance [here](https://github.com/spotify/backstage/issues/2574).
 5. The annotation value should be shortened to `repo-owner/repo-name`.

If you're looking for visual guidance for widget layout, please try to generally follow [what Iain has done for GitHub actions](https://github.com/spotify/backstage/pull/2563). The concept of a Latest Build widget is a good one for sure. It should look like Iain's design though.

File: src/components/LastBuildCard.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    </>
+  );
+};
+
+export const LastBuildCard = () => {
+  const [{ value, loading }] = useBuilds();
+  return (
+    <InfoCard title={`Latest Travis-CI Build`}>
+      {value && value.length > 0 ? (
+        <WidgetContent loading={loading} lastRun={value[0] ?? {}} />
```
Comment (iain-b) [inline src/components/LastBuildCard.tsx]: Could loading ever be true if we require length to be greater than 0?

File: src/components/LastBuildCard.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  loading,
+  lastRun,
+}: {
+  loading?: boolean;
+  lastRun: any;
+}) => {
+  const classes = useStyles();
+  if (loading || !lastRun) return <LinearProgress />;
+
+  return (
```
Comment (iain-b) [inline src/components/LastBuildCard.tsx]: What do you think about using a StructuredMetadataTable component to display this information?
https://backstage.io/storybook/?path=/story/structured-metadata-table--default
Similar usages:
plugins/jenkins/src/components/Cards/Cards.tsx:43
plugins/github-actions/src/components/Cards/Cards.tsx:59

File: src/api/index.ts [hunk 3]
```diff
@@ -77,59 +70,71 @@ export type TravisCIBuildResponse = {
...
-      await fetch(
-        `https://api.${travisVersion}/repo/${repoSlug}/builds?offset=${offset}&limit=${limit}`,
-        {
-          headers: createHeaders(token),
-        },
-      )
-    ).json();
+  async getBuilds({ limit = 10, offset = 0, repoSlug }: FetchParams) {
+    const response = await fetch(
+      `${this.baseUrl}${API_BASE_URL}/repo/${encodeURIComponent(
```
Comment (tomek-ifm) [inline src/api/index.ts]: Should thus be both this.baseUrl and API_BASE_URL here?

Comment (mcalus3) [reply src/api/index.ts]: In CircleCI backend url and base url are combined at the constructor, I think that I'll do the same here.

File: src/components/LastBuildCard.tsx (hunk 1 above)
Comment (mcalus3) [reply src/components/LastBuildCard.tsx]: Now using componen 'Table' from backstage-core

File: src/components/App.tsx [hunk 4]
```diff
@@ -1,36 +1,25 @@
...
+      <Builds />
+    </ContextProvider>
   );
 };
 
-// TODO: allow pass in settings as props
-// When some shared settings workflow
-// will be established
-export const TravisCIWidget = () => {
+export const TravisCIWidget: React.FC<Props> = ({ entity }) => {
```
Comment (iain-b) [inline src/components/App.tsx]: Can we extract this to its own file? And give the component a more specific name ? Like RecentTravisCIBuildsWidget or something to that effect if you think that's too long.

File: src/components/App.tsx [hunk 5]
```diff
@@ -1,36 +1,25 @@
...
+import { Builds } from '../components/BuildsPage';
+import { ContextProvider } from './ContextProvider';
+import { LastBuildCard } from './LastBuildCard';
 
-export const App = () => {
+type Props = {
+  entity: Entity;
+};
+
+export const App: React.FC<Props> = ({ entity }) => {
```
Comment (iain-b) [inline src/components/App.tsx]: We should rename this file & component to something more specific. TravisCIBuildsTable ? TravisCIBuildsPage

**Review by iain-b — CHANGES_REQUESTED** [review]

File: src/api/index.ts (hunk 3 above)
Comment (martina-if) [reply src/api/index.ts]: silly question: no need for an extra `/` in between?

Comment (mcalus3) [reply src/api/index.ts]: That was an outdated "experiment", it would concatenate it to "https://api.travis-ci.com/repo..."

**Pull request merged**